### PR TITLE
Implement EmployeeController, EmployeeProfiles. Create Employee model…

### PR DESCRIPTION
**Pull request by PaShkadun**
…s. Connect services and repositories in configure.

### CODE REVIEW

File: AIS/AIS.API/Controllers/EmployeeController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        public async Task<IEnumerable<EmployeeModel>> GetAll()
+        {
+            return _mapper.Map<IEnumerable<EmployeeModel>>(await _service.Get());
+        }
+
+        [HttpPost]
+        [Route("add")]
+        public async Task<EmployeeModel> Add([FromBody] EmployeeModel model)
+        {
+            return _mapper.Map<EmployeeModel>(await _service.Add(_mapper.Map<EmployeeDto>(model)));
```
Comment (VadimZharikov) [inline AIS/AIS.API/Controllers/EmployeeController.cs:34]: reafactor

File: AIS/AIS.API/Controllers/EmployeeController.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        public async Task Delete([FromBody] EmployeeModel model)
+        {
+            await _service.Delete(_mapper.Map<EmployeeDto>(model));
+        }
+
+        [HttpPut]
+        [Route("update")]
+        public async Task<EmployeeModel> Update([FromBody] EmployeeModel model)
+        {
+            return _mapper.Map<EmployeeModel>(await _service.Put(_mapper.Map<EmployeeDto>(model)));
```
Comment (VadimZharikov) [inline AIS/AIS.API/Controllers/EmployeeController.cs:48]: refactor

File: AIS/AIS.API/Controllers/EmployeeController.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        [HttpPost]
+        [Route("add")]
+        public async Task<EmployeeModel> Add([FromBody] EmployeeModel model)
+        {
+            return _mapper.Map<EmployeeModel>(await _service.Add(_mapper.Map<EmployeeDto>(model)));
+        }
+
+        [HttpDelete]
+        [Route("delete")]
```
Comment (VadimZharikov) [inline AIS/AIS.API/Controllers/EmployeeController.cs:38]: [HttpDelete("delete")] would be cleaner

File: AIS/AIS.API/Controllers/EmployeeController.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        [HttpGet]
+        public async Task<IEnumerable<EmployeeModel>> GetAll()
+        {
+            return _mapper.Map<IEnumerable<EmployeeModel>>(await _service.Get());
+        }
+
+        [HttpPost]
+        [Route("add")]
```
Comment (VadimZharikov) [inline AIS/AIS.API/Controllers/EmployeeController.cs:31]: [HttpPost("add")]

File: AIS/AIS.API/Controllers/EmployeeController.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        [HttpDelete]
+        [Route("delete")]
+        public async Task Delete([FromBody] EmployeeModel model)
+        {
+            await _service.Delete(_mapper.Map<EmployeeDto>(model));
+        }
+
+        [HttpPut]
+        [Route("update")]
```
Comment (VadimZharikov) [inline AIS/AIS.API/Controllers/EmployeeController.cs:45]: [HttpPut("update")]

File: AIS/AIS.API/Startup.cs [hunk 6]
```diff
@@ -27,7 +29,19 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            var mappingProfile = new MapperConfiguration(map =>
```
Comment (VadimZharikov) [inline AIS/AIS.API/Startup.cs:32]: add assembly mapping profiles via DI

File: AIS/AIS.API/Startup.cs [hunk 7]
```diff
@@ -27,7 +29,19 @@ public Startup(IConfiguration configuration)
...
+                map.AddProfile(new EmployeeModelProfile());
+            });
+            IMapper mapper = mappingProfile.CreateMapper();
+
+            services.AddScoped(typeof(IGenericService<>), typeof(GenericService<>));
+            services.AddScoped(typeof(IGenericRepository<>), typeof(GenericRepository<>));
+            
+            DataAccessRegister.AddDataAccess(services, Configuration);
 
+            services.AddSingleton(mapper);
```
Comment (VadimZharikov) [inline AIS/AIS.API/Startup.cs:44]: you can add mapper via DI package

File: AIS/AIS.API/ViewModels/EmployeeModel.cs [hunk 8]
```diff
@@ -0,0 +1,8 @@
+﻿namespace AIS.API.ViewModels
+{
+    public class EmployeeModel
```
Comment (nantonov) [inline AIS/AIS.API/ViewModels/EmployeeModel.cs:3]: Rename as EmployeeViewModel

File: AIS/AIS.BLL/Models/EmployeeDto.cs [hunk 9]
```diff
@@ -0,0 +1,8 @@
+﻿namespace AIS.BLL.Models
+{
+    public class EmployeeDto
```
Comment (nantonov) [inline AIS/AIS.BLL/Models/EmployeeDto.cs:3]: Rename to Employee

File: AIS/AIS.BLL/Mappers/EmployeeDtoProfile.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
+﻿using AIS.BLL.Models;
+using AIS.DAL.Entities;
+using AutoMapper;
+
+namespace AIS.BLL.Mappers
+{
+    public class EmployeeDtoProfile : Profile
```
Comment (nantonov) [inline AIS/AIS.BLL/Mappers/EmployeeDtoProfile.cs:7]: Get rid of DTO in name

File: AIS/AIS.DAL/Entities/Employee.cs [hunk 11]
```diff
@@ -0,0 +1,8 @@
+﻿namespace AIS.DAL.Entities
+{
+    public class Employee
```
Comment (nantonov) [inline AIS/AIS.DAL/Entities/Employee.cs:3]: Rename as EmployeeEntity

File: AIS/AIS.API/Controllers/EmployeeController.cs [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+        private readonly IMapper _mapper;
+        private readonly IGenericService<Employee> _service;
+
+        public EmployeeController(IMapper mapper, IGenericService<Employee> service)
+        {
+            _mapper = mapper;
+            _service = service;
+        }
+
+        [HttpGet()]
```
Comment (nantonov) [inline AIS/AIS.API/Controllers/EmployeeController.cs:25]: Maybe without ()?

File: AIS/AIS.API/Startup.cs [hunk 13]
```diff
@@ -27,6 +27,10 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddAutoMapper(typeof(EmployeeViewModelProfile), typeof(EmployeeProfile));
+
+            services.AddScoped(typeof(IGenericService<>), typeof(GenericService<>));
+            services.AddScoped(typeof(IGenericRepository<>), typeof(GenericRepository<>));
```
Comment (nantonov) [inline AIS/AIS.API/Startup.cs:33]: Why Repository is here?

File: AIS/AIS.API/Startup.cs [hunk 14]
```diff
@@ -27,6 +27,10 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddAutoMapper(typeof(EmployeeViewModelProfile), typeof(EmployeeProfile));
+
+            services.AddScoped(typeof(IGenericService<>), typeof(GenericService<>));
```
Comment (nantonov) [inline AIS/AIS.API/Startup.cs:32]: Why Service is here?

**Review by nantonov — CHANGES_REQUESTED** [review]

**Review by Blackolf931 — APPROVED** [review]

**Review by VadimZharikov — APPROVED** [review]

**Pull request merged**